controller/area_blackbox_test.go

```diff
@@ -60,190 +59,204 @@ func (rest *TestAreaREST) UnSecuredController() (*goa.Service, *AreaController)
-}
-
-func (rest *TestAreaREST) TestFailShowAreaNotFound() {
-	// given
-	svc, ctrl := rest.SecuredController()
-	// when/then
-	test.ShowAreaNotFound(rest.T(), svc.Context, svc, ctrl, uuid.NewV4().String(), nil, nil)
-}
-
-func (rest *TestAreaREST) TestShowAreaOK() {
-	// given
-	_, a := createSpaceAndArea(rest.T(), rest.db)
-	svc, ctrl := rest.SecuredController()
-	// when
-	res, _ := test.ShowAreaOK(rest.T(), svc.Context, svc, ctrl, a.ID.String(), nil, nil)
-	//then
-	assertResponseHeaders(rest.T(), res)
-}
+func (rest *TestAreaREST) TestCreateChildArea() {
+	rest.T().Run("Success", func(t *testing.T) {
+		t.Run("OK", func(t *testing.T) {
+			// given
+			fxt := tf.NewTestFixture(t, rest.DB, tf.Identities(2), tf.Areas(1))
+			parentArea := fxt.Areas[0]
+			parentID := parentArea.ID
+			ca := newCreateChildAreaPayload("TestSuccessCreateChildArea")
+			owner := fxt.Identities[0]
+			svc, ctrl := rest.SecuredControllerWithIdentity(owner)
+			// when
+			resp, created := test.CreateChildAreaCreated(t, svc.Context, svc, ctrl, parentID.String(), ca)
+			// then
+			assert.Equal(t, *ca.Data.Attributes.Name, *created.Data.Attributes.Name)
+			assert.Equal(t, parentID.String(), *created.Data.Relationships.Parent.Data.ID)
+			compareWithGoldenAgnostic(t, filepath.Join(rest.testDir, "create", "ok.res.golden.json"), created)
+			compareWithGoldenAgnostic(t, filepath.Join(rest.testDir, "create", "ok.res.headers.golden.json"), resp.Header())
+			// try creating child area with different identity: should fail
+			otherIdentity := fxt.Identities[1]
+			svc, ctrl = rest.SecuredControllerWithIdentity(otherIdentity)
+			resp, err := test.CreateChildAreaForbidden(t, svc.Context, svc, ctrl, parentID.String(), ca)
+			compareWithGolden(t, filepath.Join(rest.testDir, "create", "forbidden.res.golden.json"), err)
+			compareWithGolden(t, filepath.Join(rest.testDir, "create", "forbidden.res.headers.golden.json"), resp.Header())
+		})
+
+		t.Run("Multiple Children", func(t *testing.T) {
+			/*
+				TestAreaREST ---> TestSuccessCreateMultiChildArea-0 ----> TestSuccessCreateMultiChildArea-0-0
+			*/
+			// given
+			fxt := tf.NewTestFixture(t, rest.DB, tf.Areas(1))
+			parentArea := fxt.Areas[0]
+			parentID := parentArea.ID
+			ca := newCreateChildAreaPayload("TestSuccessCreateMultiChildArea-0")
+			owner := fxt.Identities[0]
+			svc, ctrl := rest.SecuredControllerWithIdentity(owner)
+			// when
+			resp, created := test.CreateChildAreaCreated(t, svc.Context, svc, ctrl, parentID.String(), ca)
+			// then
+			assert.Equal(t, *ca.Data.Attributes.Name, *created.Data.Attributes.Name)
+			assert.Equal(t, parentID.String(), *created.Data.Relationships.Parent.Data.ID)
+			compareWithGoldenAgnostic(t, filepath.Join(rest.testDir, "create", "ok.child1_ok.res.golden.json"), created)
+			compareWithGoldenAgnostic(t, filepath.Join(rest.testDir, "create", "ok.res.headers.golden.json"), resp.Header())
+			// Create a child of the child created above.
```

Comment:
This could be a subtest of the "ok" section and then you can get rid of line 88-104 essentially.

Reply:
Please see https://github.com/fabric8-services/fabric8-wit/pull/1941/commits/9444077af47ac799ed6063e48fb0174b1cf9c526